typescript.js

```diff
@@ -0,0 +1,408 @@
+        "@typescript-eslint/prefer-reduce-type-parameter": "warn", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-reduce-type-parameter.md
+        "@typescript-eslint/prefer-regexp-exec": "off", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-regexp-exec.md
+        "@typescript-eslint/prefer-string-starts-ends-with": "warn", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-string-starts-ends-with.md
+        "@typescript-eslint/prefer-ts-expect-error": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-ts-expect-error.md
+        "@typescript-eslint/promise-function-async": [
+            "warn",
+            {
+                allowAny: true,
+                allowedPromiseNames: [],
+                checkArrowFunctions: true,
+                checkFunctionDeclarations: true,
+                checkFunctionExpressions: true,
+                checkMethodDeclarations: true,
+            },
+        ], // https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin
+        // TODO: check @timidak
+        // def. needs testing
+        "@typescript-eslint/quotes": ["warn", "backtick", {
+            avoidEscape: true,
+        },
+        ], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/quotes.md
+        "@typescript-eslint/require-await": "warn", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/require-await.md
+        // TODO: check @timidak
+        "@typescript-eslint/return-await": ["warn", "in-try-catch"], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/return-await.md
+        // default is always
+        "@typescript-eslint/semi": ["warn"], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/semi.md
+        "@typescript-eslint/space-before-function-paren": [
+            "warn",
+            {
+                anonymous: "always",
+                named: "never",
+            },
+        ], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/space-before-function-paren.md
+        "@typescript-eslint/switch-exhaustiveness-check": "warn", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/switch-exhaustiveness-check.md
+        // i don't really think we do this anyway so let's ban it outright
+        "@typescript-eslint/triple-slash-reference": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/triple-slash-reference.md
```

Comment:
I guess create-react-app does this but not sure. Just leave it and we will go for it another time.